packages/office-ui-fabric-react/src/components/TextField/TextField.tsx

```diff
@@ -155,7 +154,7 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
     return (
       <div className={ textFieldClassName }>
         <div className={ css('ms-TextField-wrapper', styles.wrapper) }>
-          { label && <Label htmlFor={ this._id }>{ label }</Label> }
+          { this._onRenderLabel(this.props) }
```

Comment:
This will be a little confusing. Typically if we have a prop of onRenderLabel, we expect it to replace the current implementation of onRenderLabel. And the [IRenderFunction](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/utilities/src/IRenderFunction.ts#L7) should accept (props, defaultFunction). 

Take look at the [Panel](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Panel/Panel.tsx#L164)

I'd suggest having the onRenderLabel replace the entire Label element, and mix the ID into the props,

or change your onRenderLabel function to onRenderLabelGroup and not use the IRenderFunction as you probably only need `onRenderLabel?: (props) => JSX.Element`

Reply:
Thanks, after looking at Panel I see the pattern a bit more clearer now. I pushed an update which is hopefully close to what you are suggesting :) I reused the name componentId from Panel for the props, but please suggest a more fitting name if componentId is insufficient.